polaris-react/src/components/Navigation/components/Item/Item.tsx

```diff
@@ -169,18 +181,23 @@ export function Item({
       <div className={styles.Badge}>{badgeMarkup}</div>
     );
 
+  const itemLabelMarkup = (
```

Comment:
I believe I tried the truncate mixin originally, the issue is the _kind_ of truncation we're doing here, since we're using the new webkit line-clamp method of truncation so we don't need to make so many html structure changes. It also solves a lot of bugs that existed with the old "overflow: ellipses" method! I guess we could always update those components to start using the webkit-line clamp however? I'm not sure if its worth the lift though or if we'd prefer the line-clamp method everywhere.

Reply:
Thanks for all this context! I think adding it to `Truncate` (or `Text`) eventually would awesome. I've approved the UI tests so all the checks should be passing again, I think you'll just need to rebase since the other Nav item PR shipped yesterday and your good to 🚢 🚀 :shipit:

Reply:
Of course anytime and yeah for sure! Alright awesome, after some CI we will have it in!